code/mission/missionmessage.cpp

```diff
@@ -783,6 +783,56 @@ void parse_msgtbl(const char* filename)
 	}
 }
 
+int parse_message_type() {
+	char name[NAME_LENGTH];
+	stuff_string(name, F_NAME, NAME_LENGTH);
+	int type = get_builtin_message_type(name);
+	if (type == MESSAGE_NONE && stricmp(name, "None")) {
+		Warning(LOCATION, "Unknown message type %s", name);
+	}
+	return type;
+}
+
+int parse_message_priority() {
+	// TODO: Convert this to required_string_one_of.
+	if (optional_string("High")) {
+		return MESSAGE_PRIORITY_HIGH;
+	} else if (optional_string("Low")) {
+		return MESSAGE_PRIORITY_LOW;
+	} else {
+		required_string("Normal");
+		return MESSAGE_PRIORITY_NORMAL;
+	}
+}
+
+void parse_custom_message_table(const char* filename) {
```

Comment:
Why, you may ask, is this its own table, rather than a section within `messages.tbl`? Well, `messages.tbl` isn't actually parsed until a mission is loaded; conversely, LuaAI, a consumer of these message types, is loaded during startup. Loading `messages.tbl` earlier ended up having a cascading effect on the load order due to message filters. I know it's possible to load only part of a table, but I couldn't get it working the way I wanted it to. I'd certainly be willing to go that route if someone could point me in the right direction.